pkgs/servers/computing/slurm/default.nix

```diff
@@ -76,12 +91,52 @@ stdenv.mkDerivation rec {
 
 
   preConfigure = ''
-    patchShebangs ./doc/html/shtml2html.py
-    patchShebangs ./doc/man/man2html.py
+   # We're going to move the internal libraries that cause cyclic dependencies to $out,
+   # and only keep things like libpmi2 in $lib:
+   export NIX_LDFLAGS="-rpath ''${!outputLib}/lib/slurm -rpath ''${!outputBin}/lib/slurm ''${NIX_LDFLAGS-}"
+
+    while IFS= read -r -d ''$'\0' path ; do
+      patchShebangs "$path"
+    done < <( find -print0 -iname '*.py' -or -iname '*.sh')
   '';
 
-  postInstall = ''
-    rm -f $out/lib/*.la $out/lib/slurm/*.la
+  makeFlags = [
+    "perlpath=${lib.getExe buildPackages.perl}"
+  ];
+
+  installTargets = [
+    "install-recursive"
+    "install-contrib" # libpmi2
+    "PAM_DIR=${placeholder "out"}/lib/security"
+  ];
+
+  # At the time of writing, the primary motivation for splitting out $lib is to
+  # expose `libpmi2.so`, which we do. Several of the produced DSOs, however,
+  # contain references to $out (the absolute paths to $out/bin/srun, the path
+  # to $out/sbin, etc). These are cyclic references which Nix refuses to put in
+  # the store, so we move them back to $out.
+  outputBinReferences = [
+    "lib/libpmi.so*"
+    "lib/libslurm.so*"
+    "lib/slurm/libslurmfull.so*"
+    "lib/slurm/libslurm_pmi.so*"
+    "lib/slurm/mpi_pmi2.so*"
+  ];
+
+  # Note that the libtool (.la) files contain a ton of references, but we put
+  # them in a separate output so they don't affect the $out's runtime closure
+  # size
+  preFixup = ''
+    moveToOutput 'lib/slurm' "''${!outputBin}"
+    while read -r -d' ' pattern || [[ -n "$pattern" ]] ; do
+      echo "moveToOutput \"$pattern\" \"''${!outputBin}\"" >&2
+      moveToOutput "$pattern" "''${!outputBin}"
+    done <<< "$outputBinReferences"
+
+    moveToOutput '**/*.la' "''${!outputDev}"
```

Comment:
You can remove the *.la files. They are not needed to best of my knowledge.

Reply:
AFAIU, they're an archaic autoconf-adjacent analogue of `.pc`. The upstream code literally says "please do not remove", I also see no need it removing them so... why?